### PR TITLE
[IMP] account_peppol: remove peppol demo mode selection ui

### PR DESCRIPTION
**Pull request by antonrom1**
The demo mode option in the Peppol connection wizard is primarily used by internal teams (BA, support, sales) for demonstrations or testing. It is more appropriate to manage this mode through a system parameter instead of exposing it to all users in the wizard interface.

This commit removes edi selection from the settings wizard and removes related (and no longer needed) edi mode constraint as it was specific to edi mode selection ui. `account_peppol_demo.xml` sets `account_peppol.edi.mode` to `demo`, so runbot will run with peppol mode already setup to demo.

upgrade: odoo/upgrade#6594

stable pr: odoo/odoo#183180

task-4215904

### CODE REVIEW

**Review by smetl — APPROVED** [review]

File: addons/account_peppol/tools/demo_utils.py [hunk 1]
```diff
@@ -155,14 +155,11 @@ def _mock_deregister_participant(func, self, *args, **kwargs):
...
     ]).unlink()
 
-    mode_constraint = self.env['ir.config_parameter'].get_param('account_peppol.mode_constraint')
     if 'account_peppol_edi_user' in self._fields:
         self.account_peppol_edi_user.unlink()
     else:
         self.edi_user_id.unlink()
     self.account_peppol_proxy_state = 'not_registered'
-    if 'account_peppol_edi_mode' in self._fields:
-        self.account_peppol_edi_mode = mode_constraint
```
Comment (smetl) [inline addons/account_peppol/tools/demo_utils.py:165]: I think you still have to put a value in this field. When you deregister, I guess we should go back to `demo`

File: addons/account_peppol/wizard/peppol_registration_views.xml [hunk 2]
```diff
@@ -34,28 +34,6 @@
                         <field name="peppol_endpoint" string="Endpoint"/>
                         <field name="contact_email" string="Email"/>
 
-                        <!-- edi mode selection -->
-                        <field name="edi_mode_constraint" invisible="1"/>
-                        <div colspan="2" class="row" invisible="account_peppol_proxy_state != 'not_registered'">
-                            <label for="edi_mode" string="EDI mode" class="text-900 col-2"/>
-                            <field name="edi_mode"
-                                   widget="account_peppol_radio_field"
```
Comment (smetl) [inline addons/account_peppol/wizard/peppol_registration_views.xml:42]: the widget `account_peppol_radio_field` can be removed as well

**Comment by antonrom1** [comment]
@smetl fixed everything! Can we r+?

**Comment by smetl** [comment]
@robodoo r+

**Pull request closed without merging**